x/incentive/keeper/rewards.go

```diff
@@ -0,0 +1,255 @@
+package keeper
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	cdptypes "github.com/kava-labs/kava/x/cdp/types"
+	"github.com/kava-labs/kava/x/incentive/types"
+)
+
+// IterateRewardPeriods iterates over all reward period objects in the store and preforms a callback function
+func (k Keeper) IterateRewardPeriods(ctx sdk.Context, cb func(rp types.RewardPeriod) (stop bool)) {
+	store := prefix.NewStore(ctx.KVStore(k.key), types.RewardPeriodKeyPrefix)
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var rp types.RewardPeriod
+		k.cdc.MustUnmarshalBinaryLengthPrefixed(iterator.Value(), &rp)
+		if cb(rp) {
+			break
+		}
+	}
+}
+
+// HandleRewardPeriodExpiry deletes expired RewardPeriods from the store and creates a ClaimPeriod in the store for each expired RewardPeriod
+func (k Keeper) HandleRewardPeriodExpiry(ctx sdk.Context, rp types.RewardPeriod) {
+	k.CreateClaimPeriod(ctx, rp.Denom, rp.ClaimEnd, rp.ClaimTimeLock)
+	store := prefix.NewStore(ctx.KVStore(k.key), types.RewardPeriodKeyPrefix)
+	store.Delete(types.GetDenomBytes(rp.Denom))
+	return
+}
+
+// GetRewardPeriod returns the reward period from the store for the input denom and a boolean for if it was found
+func (k Keeper) GetRewardPeriod(ctx sdk.Context, denom string) (types.RewardPeriod, bool) {
```

Comment:
@rhuairahrighairidh and I have been defining base store methods that operate directly upon the store (Get, Set/Insert, Delete/Remove, Iterate, etc.) in keeper.go and defining custom methods (such as CreateNewRewardPeriod, AddToClaim, CreateAndDeleteRewardPeriods) in named keeper files. I like the standard because the base store methods are pretty universal and feel that colocating them kinda clutters the module's unique keeper functionality. See related [discussion](https://github.com/Kava-Labs/kava/pull/370#discussion_r383744618) from a previous PR.

Interested in your thoughts.